src/hyperion/parameters/plan_specific/panda/panda_gridscan_internal_params.py

```diff
@@ -48,6 +49,12 @@ def _preprocess_experiment_params(
         cls,
         experiment_params: dict[str, Any],
     ):
+
+        # Panda not configured to run a half complete snake so enforce even rows on first grid
+        # See https://github.com/DiamondLightSource/hyperion/wiki/PandA-constant%E2%80%90motion-scanning#motion-program-summary
+        if experiment_params["y_steps"] % 2:
```

Comment:
Is this only important for a 3D grid scan? If so we should check that too.

Reply:
Good catch thanks